File: server/src/check_users.ts
import dotenv from "dotenv";
import path from "path";
import { fileURLToPath } from "url";
import { db } from "./lib/postgres.js";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

dotenv.config({ path: path.join(__dirname, "../../.env") });

async function main() {
  const { data: users, error } = await db.from("users").select("id, name, email, role");
  if (error) {
    console.error("Error fetching users:", error);
  } else {
    console.log("Seeded Users in DB:", users);
  }
}

main();

File: server_py/scripts/check_users.py
import os
import sys
import asyncio

# Setup import path
base_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
sys.path.append(base_dir)

from app.db import db

async def main():
    print("Fetching all seeded users from database...")
    res = await db.from_("users").select("id, name, email, role")
    if res.error:
        print("Error fetching users:", res.error.message)
    else:
        users = res.data or []
        print(f"Found {len(users)} user(s):")
        for u in users:
            print(f"  - ID: {u['id']} | Name: {u['name']} | Email: {u['email']} | Role: {u['role']}")

if __name__ == "__main__":
    asyncio.run(main())
